services/users/src/routes/company.ts

```diff
@@ -42,8 +61,76 @@ companyRoutes.route("/:id").get(
 companyRoutes.route("/:id").put(
   checkAbility("update", "Company"),
   asyncHandler(async (req, res) => {
-    const updatedCompany = await CompanyModel.findByIdAndUpdate(req.params.id, req.body, {
-      new: true,
+    const requestedCompany = await CompanyModel.findById(req.params.id);
+
+    if (!requestedCompany) {
+      throw new BadRequestError("Requested company not valid");
+    }
+
+    const userCompany = await apiCall(
+      Service.USERS,
+      { method: "GET", url: `/companies/employees/${req.user?.uid}` },
+      req
+    );
+
+    if (!requestedCompany || !userCompany || requestedCompany.name !== userCompany.name) {
+      throw new BadRequestError("Current user not associated with requested company");
+    }
+
+    const update = await requestedCompany.update(req.body, { new: true });
+    return res.status(200).send(update);
+  })
+);
+
+companyRoutes.route("/:id").delete(
+  checkAbility("delete", "Company"),
+  asyncHandler(async (req, res) => {
+    await CompanyModel.findByIdAndDelete(req.params.id);
+    return res.sendStatus(204);
+  })
+);
+
+// get company based on employee id provided
+companyRoutes.route("/employees/:employeeId").get(
+  checkAbility("read", "Company"),
+  asyncHandler(async (req, res) => {
+    const company = await CompanyModel.findOne({ employees: req.params.employeeId });
```

Comment:
For this route, you'll also need to add a query param for hexathon, so that you can get the company for an employee for a certain hexathon. Then, you'll need to update this request in the places where u make the service to service call.

Reply:
Do we want to allow old sponsors to see the visits from the past hexathons they sponsored?

Reply:
Yeah we should be able to, but I don’t think that’s any code change on the backend, like if u take in the hexathon as a param in my other comment.